src/onnx/parse_qlinearconcat.cpp

```diff
@@ -0,0 +1,140 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/onnx/padding.hpp>
+#include <migraphx/onnx/conv.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/onnx/broadcast_qdq.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/stringutils.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+/*
+ *********************************************************************************
+ *  Reference: see QLinearConcat in                                              *
+ *  https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md  *
+ *********************************************************************************
+
+com.microsoft.QLinearConcat
+Concatenate a list of tensors into a single tensor.All input tensors must have the same shape,
+except for the dimension size of the axis to concatenate on.
+
+Version
+This version of the operator has been available since version 1 of the 'com.microsoft' operator set.
+
+Attributes
+axis : int (required)
+Which axis to concat on
+
+Inputs (3 - ∞)
+Y_scale : TF
+Y's scale.
+
+Y_zero_point : T8
+Y's zero point.
+
+inputs (variadic, heterogeneous) : TV
+List of tensors/scale/zero_point for concatenation
+
+Outputs
+Y : T8
+Concatenated tensor
+
+Type Constraints
+T8 : tensor(uint8), tensor(int8)
+Constrain input and output types to 8 bit signed and unsigned tensors.
+
+TF : tensor(float)
+Constrain scale types to any float tensor type.
+
+TV : tensor(uint8), tensor(int8), tensor(float)
+Sequence of (Tensor, Scale, ZeroPoint) tuples. The type is sequence of (T8, TF, T8).
+*/
+
+struct parse_qlinearconcat : op_parser<parse_qlinearconcat>
+{
+    std::vector<op_desc> operators() const { return {{"QLinearConcat"}}; }
+
+    // basic type checking for QLinearConcat Operator
+    void check_inputs(const std::vector<instruction_ref>& args) const
+    {
+        auto args_size = args.size();
+        // at least 5 input tensors, can be 5, 8, 11 ...
+        if((args_size < 5) or ((args_size - 2) % 3 != 0))
+            MIGRAPHX_THROW("QLINEARCONCAT: missing inputs");
+
+        auto y_zp      = args[1];
+        auto y_zp_type = y_zp->get_shape().type();
+        if(y_zp_type != migraphx::shape::int8_type and y_zp_type != migraphx::shape::uint8_type)
+            MIGRAPHX_THROW("QLINEARCONCAT: unsupported input type");
+
+        auto t0_type = args[2]->get_shape().type();
+        if(t0_type != y_zp_type)
```

Comment:
The output type doesn't have to match the input(s) type -- They both have to be either int8 or uint8, so inputs can be int8, and output be uint8. So this comparison on line 99 should be replaced by something similar to line 95 above.. Thanks.